src/rtp_api.rs

```diff
@@ -0,0 +1,64 @@
+use std::time::Instant;
+
+use crate::media::KeyframeRequestKind;
+use crate::rtp::Ssrc;
+use crate::{Rtc, RtcError};
+
+/// Wrapper for the RTP API.
+pub struct RtpApi<'a>(pub(crate) &'a mut Rtc);
+
+/// Sender id for a RTP send stream.
+pub struct SenderId(u64);
+
+impl<'a> RtpApi<'a> {
+    pub fn new_ssrc(&mut self) -> Ssrc {
+        todo!()
+    }
+
+    /// Create a new sender for outgoing RTP.
+    ///
+    /// * The SSRC must be allocated beforehand using [`allocate_ssrc`].
+    /// * The resend RTX is optional and only necessary for video.
+    pub fn new_sender(&mut self, ssrc: Ssrc, rtx: Option<Ssrc>) -> SenderId {
+        todo!()
+    }
+
+    /// Write outgoing RTP for a sender.
+    ///
+    /// * The sender must be created using [`new_sender`].
+    /// * For wallclock see below.
+    /// * Packet is the RTP packet including header. This needs to have all the wanted extension headers.
+    ///   Specifically RtpMid, RtpStreamId and RtpRepairedStreamId should be set correctly for situations
+    ///   where the SSRC is not communicated in a side channel.
+    /// * Nackable indicates whether this packet can be nacked by the remote and resent. This is always
+    ///   false for audio, but should also be set to false when the RTP payload is a discardable temporal
+    ///   layer (such as layer L1 in VP8).
+    ///
+    /// Regarding `wallclock` and `rtp_time`, the wallclock is the real world time that corresponds to
+    /// the `MediaTime`. For an SFU, this can be hard to know, since RTP packets typically only
+    /// contain the media time (RTP time). In the simplest SFU setup, the wallclock could simply
+    /// be the arrival time of the incoming RTP data (see
+    /// [`MediaData::network_time`][crate::media::MediaData]). For better synchronization the SFU
```

Comment:
To do this you really need the SR reports I think. Maybe we need to consider having str0m emit the SR, or at least data derived from them, as events?

Reply:
that's a separate problem which doesn't belong in this api design. that comment was lifted from current docs for writer.